phpBB/viewforum.php

```diff
@@ -927,8 +930,8 @@
 		$posts_unapproved = ($row['topic_visibility'] == ITEM_APPROVED && $row['topic_posts_unapproved'] && $auth->acl_get('m_approve', $row['forum_id']));
 		$topic_deleted = $row['topic_visibility'] == ITEM_DELETED;
 
-		$u_mcp_queue = ($topic_unapproved || $posts_unapproved) ? append_sid("{$phpbb_root_path}mcp.$phpEx", 'i=queue&amp;mode=' . (($topic_unapproved) ? 'approve_details' : 'unapproved_posts') . "&amp;t=$topic_id", true, $user->session_id) : '';
-		$u_mcp_queue = (!$u_mcp_queue && $topic_deleted) ? append_sid("{$phpbb_root_path}mcp.$phpEx", 'i=queue&amp;mode=deleted_topics&amp;t=' . $topic_id, true, $user->session_id) : $u_mcp_queue;
+		$u_mcp_queue = ($topic_unapproved || $posts_unapproved) ? $controller_helper->route(($topic_unapproved ? 'approve_details' : 'unapproved_posts'), ['t' => $topic_id], true, $user->session_id) : '';
```

Comment:
```suggestion
		$u_mcp_queue = ($topic_unapproved || $posts_unapproved) ? $controller_helper->route(($topic_unapproved ? 'mcp_approve_details' : 'mcp_unapproved_posts'), ['t' => $topic_id], true, $user->session_id) : '';
```

Reply:
This should now be fixed. Was also present in `search.php`